tests/ConquestV2.spec.ts

```diff
@@ -0,0 +1,603 @@
+      ownerAddress,
+      skyweaverAssetsContract.address,
+      PERIOD_LENGTH,
+      PERIOD_MINT_LIMIT.div(5),
+      false
+    ]) as RewardFactory
+
+    // Deploy conquest V1
+    oldConquestContract = await oldConquestAbstract.deploy(ownerWallet, [
+      ownerAddress,
+      skyweaverAssetsContract.address,
+      silverFactory.address,
+      goldFactory.address,
+      ticketID
+    ]) as Conquest
+    userOldConquestContract = await oldConquestContract.connect(userSigner) as Conquest
+
+    // Deploy conquest V2
+    conquestV2Contract = await conquestV2Abstract.deploy(ownerWallet, [
+      ownerAddress,
+      oldConquestContract.address,
+      skyweaverAssetsContract.address,
+      silverFactory.address,
+      goldFactory.address,
+      ticketID
+    ]) as ConquestV2
+    userConquestV2Contract = await conquestV2Contract.connect(userSigner) as ConquestV2
+
+    // Allow conquest to print cards in the factories
+    await silverFactory.assignOwnership(conquestV2Contract.address, 1)
+    await goldFactory.assignOwnership(conquestV2Contract.address, 1)
+
+    // Assing vars
+    factory = conquestV2Contract.address
+
+    // Activate gold and silver factories and authorize them
+    await skyweaverAssetsContract.activateFactory(silverFactory.address);
+    await skyweaverAssetsContract.activateFactory(goldFactory.address);
+    await skyweaverAssetsContract.addMintPermission(silverFactory.address, silver_minRange, silver_maxRange, startTime, endTime);
+    await skyweaverAssetsContract.addMintPermission(goldFactory.address, gold_minRange, gold_maxRange, startTime, endTime);
+
+    // Let owner be a "factory" to mint silver cards to test conquest
+    await skyweaverAssetsContract.activateFactory(ownerAddress);
+    await skyweaverAssetsContract.addMintPermission(ownerAddress, ticketID, ticketID, startTime, endTime);
+
+    // Mint Ticket tokens to owner and user
+    await skyweaverAssetsContract.batchMint(ownerAddress, [ticketID], [ticketAmount] , [])
+    await skyweaverAssetsContract.batchMint(userAddress, [ticketID], [ticketAmount] , [])
+
+    // Remove owner as factory
+    await skyweaverAssetsContract.shutdownFactory(ownerAddress);
+  })
+
+  let conditions = [
+    'user never used old conquest',
+    'user has nonce 2 in old conquest',
+  ]
+
+  conditions.forEach(function(condition) { 
+    context(condition as string, () => {
+      let expectedNonce
+
+      beforeEach( async () => {
+        if (condition == condition[0]) {
+          expectedNonce = 1
+        } else {
+          expectedNonce = 3
+          await userSkyweaverAssetContract.safeBatchTransferFrom(userAddress, factory, [ticketID], [100], [], TX_PARAM)
+          await conquestV2Contract.exitConquest(userAddress, 1, [], [])
+          await userSkyweaverAssetContract.safeBatchTransferFrom(userAddress, factory, [ticketID], [100], [], TX_PARAM)
+          await conquestV2Contract.exitConquest(userAddress, 2, [], [])
```

Comment:
Shouldn't these four lines be interacting with the old contract `oldConquestContract.address`?

Reply:
Oof, yes, good catch